src/app/@ansyn/plugins/openlayers/visualizers/tools/annotations.visualizer.ts

```diff
@@ -200,9 +273,10 @@ export class AnnotationsVisualizer extends EntitiesVisualizer {
 		this.iMap.projectionService
 			.projectCollectionAccurately([feature], this.iMap)
 			.take(1)
-			.subscribe((featureCollection: FeatureCollection<GeometryObject>) => {
+			.withLatestFrom(this.annotationsLayer$)
+			.subscribe(([featureCollection, annotationsLayer]: [FeatureCollection<GeometryObject>, any]) => {
 				const [geoJsonFeature] = featureCollection.features;
-				const updatedAnnotationsLayer = <FeatureCollection<any>> { ...this.annotationsLayer };
+				const updatedAnnotationsLayer = <FeatureCollection<any>> { ...annotationsLayer };
 				updatedAnnotationsLayer.features.push(geoJsonFeature);
 				this.store$.dispatch(new SetAnnotationsLayer(updatedAnnotationsLayer));
 			});
```

Comment:
(toggleDrawInteraction function) drawInteractionHandler.on('drawend', this.onDrawEndEvent.bind(this)); is not defended with subscriptions and can cause a memory leak if map is closed